harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -539,10 +542,13 @@ def _combine_metrics_across_processes(
 
             return metrics_lists, batches_per_process
         else:
-            self.train_process_comm_worker = cast(ipc.ZMQClient, self.train_process_comm_worker)
-            self.train_process_comm_worker.barrier(
-                message=ipc.MetricsInfo(metrics=metrics, num_batches=num_batches)
+            self.train_process_comm_worker = cast(
+                ipc.ZMQBroadcastClient, self.train_process_comm_worker
+            )
+            self.train_process_comm_worker.send(
+                ipc.MetricsInfo(metrics=metrics, num_batches=num_batches)
             )
+            _ = self.train_process_comm_worker.recv()
```

Comment:
might be useful to add in a comment about why this recv is needed here and in estimatorTrial

Reply:
good idea